test/pkg/environment/common/expectations.go

```diff
@@ -924,3 +926,17 @@ func (env *Environment) GetDaemonSetOverhead(np *corev1beta1.NodePool) v1.Resour
 		return p, true
 	})...)
 }
+
+func (env *Environment) EventuallyExpectNodeClassStatusCondition(nodeClass *v1beta1.EC2NodeClass, condition string, status bool, message string) {
+	GinkgoHelper()
+
+	Eventually(func(g Gomega) {
+		temp := &v1beta1.EC2NodeClass{}
+		g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(nodeClass), temp)).To(Succeed())
+		statusCondition := temp.StatusConditions().Get(condition)
```

Comment:
nit: consider using a variable more indicative of the type
```suggestion
		nc := &v1beta1.EC2NodeClass{}
		g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(nodeClass), nc)).To(Succeed())
		statusCondition := nc.StatusConditions().Get(condition)
```